cpp/src/arrow/compute/api_scalar.h

```diff
@@ -42,6 +42,11 @@ struct ArithmeticOptions : public FunctionOptions {
   bool check_overflow;
 };
 
+struct ARROW_EXPORT MinMaxOptions : public FunctionOptions {
```

Comment:
Do we want to name this something else?

Reply:
I think it is worth giving this a more general name, because there are other potential uses. Maybe `ElementWiseAggregateOptions`? (since this is aggregation, just across instead of down)

Reply:
If using "Aggregate" is too confusing, maybe "Combine" or "Merge" would work too. e.g. `ElementWiseCombineOptions`

Reply:
Since we have `ScalarAggregateOptions` I think `ElementWiseAggregateOptions` should be ok. I've updated the PR.